dataflowengineoss/src/main/scala/io/shiftleft/dataflowengineoss/passes/reachingdef/ReachingDefPass.scala

```diff
@@ -303,6 +128,31 @@ class ReachingDefPass(cpg: Cpg) extends ParallelCpgPass[nodes.Method](cpg) {
     dstGraph
   }
 
+  private def uses(node: nodes.StoredNode, gen: Map[nodes.StoredNode, Set[nodes.StoredNode]]): Set[nodes.StoredNode] = {
+    node match {
+      case ret: nodes.Return =>
+        ret.astChildren.map(_.asInstanceOf[nodes.StoredNode]).toSet()
+      case call: nodes.Call =>
+        val parameters = NoResolve.getCalledMethods(call).headOption.map(_.parameter.l).getOrElse(List())
+        node
+          ._argumentOut()
+          .asScala
+          .filter(
+            arg =>
+              parameters
+                .filter(_.order == arg.asInstanceOf[HasOrder].order)
+                .flatMap(_._propagateOut().asScala.toList)
+                .nonEmpty || !gen(node).contains(arg))
```

Comment:
just in case you missed it, having an _and_ here would make the logic clearer; but again, nbd